src/modules/auth/store/authActions.store.ts

```diff
@@ -0,0 +1,48 @@
+import { PayloadAction, createAsyncThunk } from '@reduxjs/toolkit';
+import { signup, login } from '../api/index.api.ts';
+import { AuthState, TrainerSignup, Login } from '../types';
+
+export interface AuthStateOptions extends Partial<AuthState> {}
+
+export function updateAuthStateAction(
```

Comment:
Nice `update` fn! I like how you can practically update anything in the auth state.